src/Symfony/Component/VarDumper/Caster/XmlReaderCaster.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Casts XmlReader class to array representation
+ *
+ * @author Baptiste Clavié <clavie.b@gmail.com>
+ */
+class XmlReaderCaster
+{
+    private static $nodeTypes = array(
+            XmlReader::NONE => 'NONE',
+            XmlReader::ELEMENT => 'ELEMENT',
+            XmlReader::ATTRIBUTE => 'ATTRIBUTE',
+            XmlReader::TEXT => 'TEXT',
+            XmlReader::CDATA => 'CDATA',
+            XmlReader::ENTITY_REF => 'ENTITY_REF',
+            XmlReader::ENTITY => 'ENTITY',
+            XmlReader::PI => 'PI',
+            XmlReader::COMMENT => 'COMMENT',
+            XmlReader::DOC => 'DOC',
+            XmlReader::DOC_TYPE => 'DOC_TYPE',
+            XmlReader::DOC_FRAGMENT => 'DOC_FRAGMENT',
+            XmlReader::NOTATION => 'NOTATION',
+            XmlReader::WHITESPACE => 'WHITESPACE',
+            XmlReader::SIGNIFICANT_WHITESPACE => 'SIGNIFICANT_WHITESPACE',
+            XmlReader::END_ELEMENT => 'END_ELEMENT',
+            XmlReader::END_ENTITY => 'END_ENTITY',
+            XmlReader::XML_DECLARATION => 'XML_DECLARATION'
+    );
+
+    public static function castXmlReader(XmlReader $reader, array $a, Stub $stub, $isNested)
+    {
+        $infos = array(
+            'name' => $reader->name,
+            'local_name' => $reader->localName,
+
+            'depth' => $reader->depth,
+
+            'attributes_count' => $reader->attributeCount,
+            'has_attributes' => $reader->hasAttributes,
+
+            'has_value' => $reader->hasValue,
+            'is_default' => $reader->isDefault,
+            'is_empty' => $reader->isEmptyElement,
+            'node_type' => new ConstStub(self::$nodeTypes[$reader->nodeType], $reader->nodeType)
+        );
+
+        if ($reader->hasValue && XmlReader::TEXT === $reader->nodeType) {
+            $infos['value'] = new CutStub($reader->value);
+        }
+
+        if ($reader->hasAttributes) {
+            $infos['attributes'] = array();
+
+            while ($reader->moveToNextAttribute()) {
```

Comment:
this changes the state of the reader, which is a no-go. The casters are not allowed to produce side effects.